### ISSUE TITLE
Simultaneous delete and create on the same node can cause long delays

### ISSUE THREAD

**Issue by matthias50** [cri-o/cri-o#3651]
**Description**

When I delete  Pods with a particular combination of init containers and "regular" containers, while at the same time attempting to create the same particular combination, the second set gets delayed while the first set is in the Terminating state.

**Steps to reproduce the issue:**

1. Add a label to a node so we can control where pods get scheduled 
```
kubectl label node <redacted> workload=batch-test
```
2. Create namespaces for running the identical Pods
```
kubectl create namespace test
```
3. create one copy
```
kubectl -n test apply -f pod_example-1.yaml
kubectl -n test get pods
```
4. delete the first copy and very shortly thereafter, create the second
```
kubectl -n test delete -f pod_example-1.yaml&
sleep 1
kubectl create -n test -f pod_example-2.yaml&
```
5. Watch the pods to observe that the second set gets stuck in the Init state till the first set is Terminated
```
kubectl -n test get pods -w
```

**Describe the results you received:**
The Pod batch-test-2 took over 2 minutes to transition to Running

Output:
```
$ kubectl create namespace test
namespace/test created
$ kubectl -n test apply -f pod_example-1.yaml
pod/batch-test-1 created
$ kubectl -n test get pods
NAME           READY   STATUS     RESTARTS   AGE
batch-test-1   0/2     Init:0/1   0          1s
$ kubectl -n test delete -f pod_example-1.yaml&
[1] 1296
$ sleep 1
pod "batch-test-1" deleted
$ kubectl create -n test -f pod_example-2.yaml&
pod/batch-test-2 created
[2] 1309
$ kubectl -n test get pods -w

NAME           READY   STATUS        RESTARTS   AGE
batch-test-1   2/2     Terminating   0          5s
batch-test-2   0/2     Init:0/1      0          2s
batch-test-2   0/2     PodInitializing   0          2m
batch-test-1   0/2     Terminating       0          2m4s
batch-test-1   0/2     Terminating       0          2m4s
batch-test-2   2/2     Running           0          2m1s
batch-test-1   0/2     Terminating       0          2m12s
batch-test-1   0/2     Terminating       0          2m12s
```
**Describe the results you expected:**
The Pod batch-test-2 transitions to running within a few seconds.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `crio --version`:**

```
crio version 1.15.3-1.el7
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
AWS 
```
$ cat /etc/os-release
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"
```

Contents of pod_example-1.yaml
```
kind: Pod
apiVersion: v1
metadata:
  name: batch-test-1
spec:
  terminationGracePeriodSeconds: 120
  nodeSelector:
    workload: "batch-test"
  initContainers:
  - name: init-mydb
    image: busybox:1.28
    command: ['sh', '-c', "ls"]
  containers:
  - name: worker
    image: k8s.gcr.io/nginx-slim:0.8
    ports:
    - containerPort: 80
      name: web
  - name: executor
    image: busybox:1.28
    command: ['sh', '-c', "sleep 20m"]
```
Contents of pod_example-2.yaml
```
kind: Pod
apiVersion: v1
metadata:
  name: batch-test-2
spec:
  terminationGracePeriodSeconds: 120
  nodeSelector:
    workload: "batch-test"
  initContainers:
  - name: init-mydb
    image: busybox:1.28
    command: ['sh', '-c', "ls"]
  containers:
  - name: worker
    image: k8s.gcr.io/nginx-slim:0.8
    ports:
    - containerPort: 80
      name: web
  - name: executor
    image: busybox:1.28
    command: ['sh', '-c', "sleep 20m"]
```

**Comment by mrunalp** [comment]
Are you able to test this with `release-1.17` or `release-1.16`?

**Comment by haircommander** [comment]
@matthias50 https://github.com/cri-o/cri-o/pull/3701 fixed it for me. can you test it out too?

**Comment by haircommander** [comment]
seems to be a dup of https://github.com/cri-o/cri-o/issues/3455

**Comment by matthias50** [comment]
@haircommander, I tried out the fix in #3701 and it worked for me. Thanks.

**Comment by haircommander** [comment]
https://github.com/cri-o/cri-o/pull/3709 has merged, closing this
